frontend/src/components/pages/StatisticsPage.tsx

```diff
@@ -0,0 +1,461 @@
+
+    filteredData.forEach((item) => {
+      const xVal = getAxisValue(item, xAttr);
+      const yVal = getAxisValue(item, yAttr);
+
+      if (!groupMap.has(xVal)) groupMap.set(xVal, new Map());
+      const yMap = groupMap.get(xVal)!;
+      if (yAttr === 'price') {
+        yMap.set('price', (yMap.get('price') ?? 0) + Number(item.price ?? 0));
+      } else {
+        yMap.set(yVal, (yMap.get(yVal) ?? 0) + 1);
+      }
+    });
+
+    const xLabels = Array.from(groupMap.keys());
+    let yLabels: string[];
+    let series;
+
+    if (yAttr === 'price') {
+      yLabels = ['price'];
+      series = [{
+        data: xLabels.map((xLabel) => groupMap.get(xLabel)?.get('price') ?? 0),
+        label: 'Сумма price',
+      }];
+    } else {
+      const yLabelsSet = new Set<string>();
+      groupMap.forEach((yMap) => {
+        yMap.forEach((_, y) => yLabelsSet.add(y));
+      });
+      yLabels = Array.from(yLabelsSet);
+      series = yLabels.map((yLabel) => ({
+        data: xLabels.map((xLabel) => groupMap.get(xLabel)?.get(yLabel) ?? 0),
+        label:
+          yAttr === 'category'
+            ? yLabel
+            : yAttr === 'timestamp'
+              ? yLabel
+              : yAttr === 'user_fullname'
+                ? yLabel
+                : yLabel,
+      }));
+    }
+
+    return { xLabels, series };
+  }, [filteredData, xAttr, yAttr, usersList]);
+
+  const userFullnameValue = filters.user_id
+    ? usersList?.[filters.user_id] || ''
+    : '';
+
+  const unique = (attr: keyof ProductRequest) =>
+    Array.from(new Set((productsData ?? []).map((item) => String(item[attr] ?? '')))).filter(Boolean);
+
+  return (
+    <Box p={4}>
+      <Button component={Link} to="/" variant="contained" sx={{ mb: 2 }}>
+        В главное меню
+      </Button>
+      <Typography variant="h5" gutterBottom>
+        Статистика по заявкам
+      </Typography>
+
+      <Box display="flex" gap={2} flexWrap="wrap">
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={onlyMy}
+              onChange={e => setOnlyMy(e.target.checked)}
+            />
+          }
+          label="Участвовал в разрешении"
+        />
+      </Box>
+
+      <Box display="flex" gap={2} flexWrap="wrap" mb={3}>
+        <TextField
+          label="Название содержит"
+          value={filters.title}
+          onChange={e => setFilters(f => ({ ...f, title: e.target.value }))}
+          size="small"
+        />
+        <FormControl size="small" style={{ minWidth: 140 }}>
+          <InputLabel>{attrToRussian['category']}</InputLabel>
+          <Select
+            value={filters.category}
+            label={attrToRussian['category']}
+            onChange={e => setFilters(f => ({ ...f, category: e.target.value }))}
+          >
+            <MenuItem value="">Все</MenuItem>
+            {unique('category').map(val => (
+              <MenuItem key={val} value={val}>{categoryToRussian(val as any) ?? val}</MenuItem>
```

Comment:
Тут можно без `any`:

```suggestion
              <MenuItem key={val} value={val}>{categoryToRussian(val as CategoryEnum) ?? val}</MenuItem>
```